controllers/vspherecluster_controller.go

```diff
@@ -274,11 +274,7 @@ func (r clusterReconciler) reconcileNormal(ctx *context.ClusterContext) (reconci
 	ctx.VSphereCluster.Status.Ready = true
 
 	// Reconcile the VSphereCluster resource's control plane endpoint.
-	if ok, err := r.reconcileControlPlaneEndpoint(ctx); !ok {
-		if err != nil {
-			return reconcile.Result{}, errors.Wrapf(err,
-				"unexpected error while reconciling control plane endpoint for %s", ctx)
-		}
+	if ok := r.reconcileControlPlaneEndpoint(ctx); !ok {
```

Comment:
This function and its logic is small enough that it can probably be inlined at this point: 
```
r.reconcileVSphereClusterWhenAPIServerIsOnline(ctx)
if ctx.VSphereCluster.Spec.ControlPlaneEndpoint.IsZero() {
  ctx.Logger.Info("control plane endpoint is not reconciled")
  return reconcile.Result{}, nil
}
```

Reply:
Ack